virttest/utils_test/qemu.py

```diff
@@ -1842,3 +1842,308 @@ def get_all_memorys(self, vm):
+
+        """
+        create and set a drive object
+        """
+        self.mkimg(params, test, image)
+        usb_params = params.object_params(image)
+        filename = storage.get_image_filename(usb_params,
+                                              data_dir.get_data_dir())
+        drive = qdevices.QRHDrive(image)
+        drive.set_param('if', interface)
+        drive.set_param('format', fmt)
+        drive.set_param('file', filename)
+        if serial:
+            drive.set_param('serial', serial)
+        if cache:
+            drive.set_param('cache', cache)
+        if snapshot:
+            drive.set_param('snapshot', snapshot)
+        return drive
+
+    def device_set(self, usb_name, driver, controller_type=None,
+                   drive='drive_%s', bus='usbtest.0', port=None):
+
+        """
+        create and set a device
+        """
+        dev = qdevices.QDevice(driver, aobject=usb_name)
+        dev.set_param('id', 'usb-%s' % usb_name)
+        if drive:
+            dev.set_param('drive', drive % usb_name)
+        dev.set_param('bus', bus)
+        if port:
+            dev.set_param('port', port)
+        if controller_type:
+            dev.parent_bus += ({'type': controller_type},)
+        return dev
+
+    def device_add(self, dev, driver, vm):
+
+        """
+        hot plug a usb device
+        """
+        logging.info("Begin to hot plug %s device" % driver)
+        out = dev.hotplug(vm.monitor)
+        if out:
+            logging.info("Hot plug %s device with error %s" %
+                         (driver, out))
+        time.sleep(3)
+        logging.info("Begin to verify %s device" % driver)
+        result = dev.verify_hotplug(out, vm.monitor)
+        if not result:
+            raise exceptions.TestFail(" %s device wasn't found in monitor" %
+                                      driver)
+        else:
+            logging.info(" %s device was found in monitor" % driver)
+
+    def device_del(self, dev, driver, vm):
+
+        """
+        hot unplug a usb device
+        """
+        logging.info("Begin to unplug %s device" % driver)
+        out = dev.unplug(vm.monitor)
+        if out:
+            logging.info("Unplug %s device with error %s" % (driver, out))
+        time.sleep(3)
+        logging.info("Begin to verify %s device" % driver)
+        result = dev.verify_unplug(out, vm.monitor)
+        if not result:
+            raise exceptions.TestFail(" %s device was found in monitor" %
+                                      driver)
+        else:
+            logging.info(" %s device wasn't found in monitor" % driver)
+
+    def usb_dev_hotplug(self, driver, repeat_times, vm):
+
+        """
+        hotplug and unplug usb device
+        """
+        logging.info("Hot plug and unplug usb device")
+        for i in xrange(repeat_times):
+            if driver == 'usb-storage':
+                drive = self.drive_set(self.params, self.test, 'udev')
```

Comment:
No need to transfer self.params and self.test, which have already been init in __init__.